source/java/src/org/lucee/extension/pdf/tag/PDF.java

```diff
@@ -1338,15 +1338,23 @@ private void doActionSetInfo() throws PageException, IOException, DocumentExcept
 
 		PDFStruct doc = toPDFDocument(source, password, null);
 		PdfReader pr = doc.getPdfReader();
-		OutputStream os = null;
+		OutputStream os = new ByteArrayOutputStream();
+
+		if (destination != null && name != null) throw engine.getExceptionUtil().createApplicationException("you cannot use attributes[destination,name] together, must specify one of them");
+
 		try {
 			if (destination == null) {
-				if (doc.getResource() == null) throw engine.getExceptionUtil().createApplicationException("Source is not based on a resource, destination file is required");
-				destination = doc.getResource();
+				if (Util.isEmpty(name)) {
+					if (doc.getResource() != null) destination = doc.getResource();
+					else if (source instanceof String && doc.getResource() == null) name = (String) source;
+					else throw engine.getExceptionUtil().createApplicationException("PDF [source] is not based on a resource or string variable value, must specify one of the following attribute [destination,name]");
```

Comment:
PDF attribute [source] is not a resource (file) or variable, you must specify one of the following attributes [ destination, name]

let's say file rather than resource, it's more cfml-y?